### PR TITLE
add C_Auction_ID to C_OLCand and populate it all the way to C_Invoice

### PR DESCRIPTION
**Pull request by adi-stefan**

### CODE REVIEW

File: backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/OLCandOrderFactory.java [hunk 1]
```diff
@@ -315,6 +321,17 @@ private I_C_Order newOrder(@NonNull final OLCand candidateOfGroup)
 		order.setEMail(candidateOfGroup.getEmail());
 		order.setPhone(candidateOfGroup.getPhone());
 		order.setM_SectionCode_ID(SectionCodeId.toRepoId(getSectionCodeId(candidateOfGroup)));
+		final AuctionId auctionId = candidateOfGroup.getAuctionId();
+		if (auctionId != null)
+		{
+			order.setC_Auction_ID(AuctionId.toRepoId(auctionId));
```
Comment (teosarca) [inline backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/OLCandOrderFactory.java:327]: imho u shall move this inside `if(action!=null)`

**Review by teosarca — CHANGES_REQUESTED** [review]
check my comments, besides that lgtm

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/order/invoicecandidate/C_OrderLine_Handler.java [hunk 2]
```diff
@@ -182,12 +182,6 @@ private I_C_Invoice_Candidate createCandidateForOrderLine(final I_C_OrderLine or
 
 		icRecord.setProductDescription(orderLine.getProductDescription());
 
-		if (orderLine.getPriceEntered().compareTo(orderLine.getPriceStd()) == 0)
```
[the remark this answers is not in the document]
Comment (adi-stefan) [reply backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/order/invoicecandidate/C_OrderLine_Handler.java:185]: the columns were removed from DB, but the classes were `C_Invoice_Candidate` classes were not regenerated.
See: https://github.com/metasfresh/metasfresh/blob/0fbd3e88370dd123dc3cb55ac17b47c9ec42cb4c/backend/de.metas.swat/de.metas.swat.base/src/main/sql/postgresql/system/47-de.metas.invoicecandidate/5604430_sys_gh11720_drop_unused_commission_cols_invoice_cand.sql

File: backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/OLCandOrderFactory.java (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.salescandidate.base/src/main/java/de/metas/ordercandidate/api/OLCandOrderFactory.java]: done.

**Review by teosarca — APPROVED** [review]

**Pull request merged**